### PR TITLE
Get OpenEye to save SEQRES headers when prepping design units

### PR DESCRIPTION
**Pull request by kaminow**
## Description
Currently the `prep_proteins.py` script doesn't save a PDB file with SEQRES headers. This PR aims to address that using a combination of specifying the `OESequenceMetadata` when creating the design units, and manually adding the SEQRES headers when saving the complex molecule.

## Todos
  - [x] Use `OESequenceMetadata` to avoid having to write an intermediate PDB file
  - [x] Add SEQRES headers to PDB file output by the `prep_proteins` script

## Discussion
I'm still not sure how to have the SEQRES info persist with the molecules. It's definitely being used in the creation of the design units, but I'm not sure where it goes afterwards. The code here works and accomplishes the goal, but there might be a more elegant/robust way hidden in OpenEye somewhere.

## Status
- [x] Ready to go

### CODE REVIEW

File: asapdiscovery/docking/scripts/prep_proteins.py [hunk 1]
```diff
@@ -148,6 +124,7 @@ def prep_mp(
             site_residue=site_residue,
             loop_db=loop_db,
             protein_only=protein_only,
+            seqres=" ".join(res_list),
```
[the remark this answers is not in the document]
Comment (apayne97) [reply asapdiscovery/docking/scripts/prep_proteins.py:127]: @kaminow unless I'm mistaken, you added the `seqres` option to the `prep_receptor` function, not the `mutate_residues` function. which means that it should be added as an argument further down at that step.

[the remark this answers is not in the document]
Comment (apayne97) [reply asapdiscovery/docking/scripts/prep_proteins.py:127]: oh, no I STILL don't know how GitHub works. disregard this whole thread, i was thrown off by the way it was represented in the review page. not my day lol.

File: notebook/test_sequence_metadata.ipynb [hunk 2]
```diff
@@ -0,0 +1,460 @@
+{
```
[the remark this answers is not in the document]
Comment (kaminow) [reply notebook/test_sequence_metadata.ipynb:1]: done in [bda1321](https://github.com/choderalab/covid-moonshot-ml/pull/116/commits/bda13212191eaa99c463eedf9127cdb9ffe6fede)

File: asapdiscovery/docking/scripts/prep_proteins.py [hunk 3]
```diff
@@ -169,6 +146,13 @@ def prep_mp(
 
         ## Save complex as PDB file
         complex_mol = du_to_complex(du, include_solvent=True)
+
+        ## Add SEQRES entries if they're not present
+        if (not oechem.OEHasPDBData(complex_mol, "SEQRES")) and seqres:
+            for seqres_line in seqres.split("\n"):
+                if seqres_line != "":
+                    oechem.OEAddPDBData(complex_mol, "SEQRES", seqres_line[6:])
```
[the remark this answers is not in the document]
Comment (kaminow) [reply asapdiscovery/docking/scripts/prep_proteins.py:154]: the header lines are pretty strictly formatted, so any correct `SEQRES` header will just have those 6 characters and then the rest of the line. I suppose a potentially more robust way would be something like
```python
oechem.OEAddPDBData(complex_mol, "SEQRES", seqres_line.split("SEQRES")[1])
```

[the remark this answers is not in the document]
Comment (apayne97) [reply asapdiscovery/docking/scripts/prep_proteins.py:154]: @kaminow @JenkeScheen I can change this in #104 but i think we're good to merge for now?

**Pull request merged**